pkg/clients/handlers.go

```diff
@@ -0,0 +1,224 @@
+package clients
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/canonical/identity-platform-admin-ui/internal/logging"
+	"github.com/canonical/identity-platform-admin-ui/internal/responses"
+	"github.com/go-chi/chi/v5"
+)
+
+type API struct {
+	service ServiceInterface
+
+	logger logging.LoggerInterface
+}
+
+type PaginationLinksResponse struct {
+	First string `json:"first,omitempty"`
+	Last  string `json:"last,omitempty"`
+	Prev  string `json:"prev,omitempty"`
+	Next  string `json:"next,omitempty"`
+}
+
+func (a *API) RegisterEndpoints(mux *chi.Mux) {
+	mux.Get("/api/v0/clients", a.ListClients)
+	mux.Post("/api/v0/clients", a.CreateClient)
+	mux.Get("/api/v0/clients/{client_id}", a.GetClient)
+	mux.Put("/api/v0/clients/{client_id}", a.UpdateClient)
+	mux.Delete("/api/v0/clients/{client_id}", a.DeleteClient)
```

Comment:
minor thing, but here i'd stick with `{id}` reason is it is simple to do regexp patterns in the middleware when the `id` name is standardized